Phase08/Phase08/Phase08/DataHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Castle.Core.Internal;
+using SQLHandler;
+
+namespace Phase08
+{
+    public class DataHandler : IDataHandler
+    {
+        private readonly IFileReader _fileReader;
+        private readonly IInvertedIndex _invertedIndex;
+        private readonly InvertedIndexContext _invertedIndexContext;
+
+        public DataHandler(IFileReader fileReader, IInvertedIndex invertedIndex,
+            InvertedIndexContext invertedIndexContext)
+        {
+            _fileReader = fileReader;
+            _invertedIndex = invertedIndex;
+            _invertedIndexContext = invertedIndexContext;
+        }
+
+        public void InitializeDataBase(string folderPath)
+        {
+            if (IsDataBaseInitialized()) return;
+            
+            var documents = _fileReader.ReadFile(folderPath);
+            _invertedIndex.BuildInvertedIndex(documents);
+        }
+
+        private bool IsDataBaseInitialized()
```

Comment:
behtare to InvertedIndexContext ye tabe bezarin ke ino beheton bege, jay inke dasti check konin

Reply:
va inke behtare roy classi mesl context ye wrapper bekeshin va faghat on ghabeliat haei ke niaz darin ro be baghie class ha bedin